File: find-the-highest-altitude/Solution1732.java
// https://leetcode.com/problems/find-the-highest-altitude/

class Solution1732 {

    // T: O(n), S: O(1)
    public int largestAltitude(int[] gain) {
        int max = 0, curr = 0;
        for (int val : gain) {
            curr += val;
            if (curr > max) {
                max = curr;
            }
        }
        return max;
    }

    public static void main(String[] args) {
        var sol = new Solution1732();
        var nums1 = new int[] { -5, 1, 5, 0, -7 };
        var nums2 = new int[] { -4, -3, -2, -1, 4, 3, 2 };
        System.out.println(sol.largestAltitude(nums1)); // 1
        System.out.println(sol.largestAltitude(nums2)); // 0
    }

}

File: find-the-highest-altitude/solution1732.go
// https://leetcode.com/problems/find-the-highest-altitude/

package main

import "fmt"

// T: O(n), S: O(1)
func largestAltitude(gain []int) int {
	max, curr := 0, 0
	for _, val := range gain {
		curr += val
		if curr > max {
			max = curr
		}
	}
	return max
}

func main() {
	nums1 := []int{-5, 1, 5, 0, -7}
	nums2 := []int{-4, -3, -2, -1, 4, 3, 2}
	fmt.Println(largestAltitude(nums1)) // 1
	fmt.Println(largestAltitude(nums2)) // 0
}
